pylsp/plugins/jedi_completion.py

```diff
@@ -81,14 +89,27 @@ def pylsp_completions(config, document, position):
     if include_class_objects:
         for c in completions:
             if c.type == 'class':
-                completion_dict = _format_completion(c, False)
+                completion_dict = _format_completion(c, False, resolve=resolve_eagerly)
                 completion_dict['kind'] = lsp.CompletionItemKind.TypeParameter
                 completion_dict['label'] += ' object'
                 ready_completions.append(completion_dict)
 
+    _LAST_COMPLETIONS = {
+        # label is the only required property; here it is assumed to be unique
+        completion['label']: (completion, data)
+        for completion, data in zip(ready_completions, completions)
+    }
+
     return ready_completions or None
 
 
+@hookimpl
+def pylsp_completion_item_resolve(completion_item):
+    """Resolve formatted completion for given non-resolved completion"""
+    completion, data = _LAST_COMPLETIONS.get(completion_item['label'])
```

Comment:
This hookspec should use the proposed cache mechanism, instead of a global variable